drivers/platform/aducm3029/flash.c

```diff
@@ -0,0 +1,250 @@
+ *    from the requirement that you obtain separate licenses from these
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/************************* Include Files **************************************/
+/******************************************************************************/
+
+#include <drivers/flash/adi_flash.h>
+#include <drivers/dma/adi_dma.h>
+#include <stdlib.h>
+#include <malloc.h>
+#include "flash.h"
+#include "flash_extra.h"
+#include "error.h"
+
+/******************************************************************************/
+/*************************** Types Declarations *******************************/
+/******************************************************************************/
+
+/**
+ * @struct adicup_flash_dev
+ * @brief Aducm3029 flash controller handler
+ */
+struct adicup_flash_dev {
+	void *instance;
+	void *dfp_memory;
+};
+
+/******************************************************************************/
+/************************ Functions Definitions *******************************/
+/******************************************************************************/
+
+/**
+ * Initialize flash controller.
+ *
+ * @param [out] device     - Pointer to the driver handler.
+ * @param [in]  init_param - Pointer to the initialization structure.
+ *
+ * @return 0 in case of success, error code otherwise.
+ */
+int32_t flash_init(struct flash_dev **device,
+		   struct flash_init_param *init_param)
+{
+	int32_t ret;
+	struct flash_dev *dev;
+	struct adicup_flash_dev *adicup_extra;
+
+	dev = (struct flash_dev *)calloc(1, sizeof(*dev));
+	if (!dev)
+		return FAILURE;
+
+	dev->id = init_param->id;
+	adicup_extra = (struct adicup_flash_dev *)calloc(1,
+			sizeof(*adicup_extra));
+	if (!adicup_extra)
+		goto error_dev;
+
+	dev->extra = adicup_extra;
+	adicup_extra->dfp_memory = calloc(1, ADI_FEE_MEMORY_SIZE);
```

Comment:
You can do:
'''
struct adicup_flash_dev {
	void *instance;
	uint8_t dfp_memory[ADI_FEE_MEMORY_SIZE]; 
};
'''
And maybe align it if needed.

Reply:
Changed